### ISSUE TITLE
The order of evaluation of environments is not consistent in sd-local

### ISSUE THREAD

**Issue by itleigns** [screwdriver-cd/screwdriver#2446]
**What happened**:
The order of evaluation of environment variables is not consistent in sd-local.
So the result of the evaluation of an environment variable is not constant when another environment variable is used in the environment variable.

**What you expected to happen**:
Environment variables are evaluated in the order in which they are defined in `screwdriver.yaml`.

**References**:
https://docs.screwdriver.cd/user-guide/configuration/environment

**How to reproduce it**:
Use the following YAML.
```
jobs:
  foo:
    image: node12
    steps:
    - echo $BEFORE
    - echo $AFTER
    environment:
      BEFORE: YYY${X}YYY
      X: XXX
      AFTER: ZZZ${X}ZZZ
```
`BEFORE` can be evaluated to either `YYYXXXYYY` or `YYYYYY` and `AFTER` can be evaluated to either `ZZZXXXZZZ` or `ZZZ`.

**Comment by tanoda** [comment]
On YAML, you should not rely order of mapping. it's an implementation detail.
https://yaml.org/spec/1.2/spec.html#id2765608

**Comment by itleigns** [comment]
Currently, Screwdriver.cd evaluates environment variables in the order in which they are defined. So for now, sd-local follows that specification. Whether the Screwdriver.cd specification is valid or not will be discussed in another issue.

**Comment by itleigns** [comment]
I created issue.
https://github.com/screwdriver-cd/screwdriver/issues/2447

**Comment by tanoda** [comment]
> I created issue.
> #2447

Thanks.
For JSON, [ECMA-404](https://www.ecma-international.org/publications-and-standards/standards/ecma-404/) says `The JSON syntax does not impose any restrictions on
the strings used as names, does not require that name strings be unique, and does not assign any
significance to the ordering of name/value pairs.`, but generally we should assume unordered set too, as [JSON.org](https://www.json.org/json-en.html) says.

**Comment by itleigns** [comment]
fixed with: https://github.com/screwdriver-cd/sd-local/pull/73
